faker/providers/misc/__init__.py

```diff
@@ -87,12 +87,17 @@ def sha256(self, raw_output=False):
         return res.hexdigest()
 
     def uuid4(self, cast_to=str):
-        """Generate a random UUID4 object and cast it to another type using a callable ``cast_to``.
+        """Generate a random UUID4 object and cast it to another type if specified using a callable ``cast_to``.
 
         By default, ``cast_to`` is set to ``str``.
+
+        May be called with ``cast_to=None`` to return a full-fledged ``UUID``.
```

Comment:
The autodoc system can now automatically create samples. If you want to learn more about how things are done under the hood, you may refer to this [page](https://faker.readthedocs.io/en/master/writing-docs.html). On that note, can you please add samples? Basically just change this line to this verbatim:

```
May be called with ``cast_to=None`` to return a full-fledged ``UUID``.

:sample:
:sample: cast_to=None
```

Reply:
That's brilliant! Didn't know it was even possible to do this